File: Domain/Business.cs
using System;
using System.Collections.Generic;
namespace Domain
{
    public class Business
    {
         public Guid Id { get; set;}
        public string Title {get; set; }
        public string Description { get; set; }
        public string FeaturedPost { get; set; }
         public string Category { get; set; }
         public string Street { get; set; }
         public string City { get; set; }
         public string State { get; set; }
         public string Website { get; set; }
         public string Image { get; set; }
         public string Hours { get; set; }
         public bool IsService { get; set; }
        public DateTime Date { get; set; }
        public virtual ICollection<UserBusiness> UserBusinesses {get; set;}
        public virtual ICollection<BusinessComment> BusinessComments {get; set;}        
          
    }
}

File: client-app/src/App/Models/business.ts
export interface IBusinessesEnvelope {
    businesses: IBusiness[];
    businessCount: number;
  }

export interface IBusiness {
    id: string;
    title: string;
    description: string;
    featuredPost: string;
    street: string;
    hours: string;
    city: string;
    image: string;
    state: string;
    website: string;
    isService: boolean;
    category: string;
    date: Date;
    isHost: boolean;
    isLiked: boolean;
    liked: ILikedBusiness[];
    businessComments: IComment[];
}

export interface IComment {
    id: string;
    createdAt: Date;
    body: string;
    username: string;
    displayName: string;
    image: string;
  }

export interface IBusinessFormValues extends Partial<IBusiness> {
    time?: Date;
}
export class BusinessFormValues implements IBusinessFormValues {
    id?: string = undefined;
    title: string = '';
    category: string = "";
    description: string = "";
    date?: Date = undefined;
    time?: Date = undefined;
    featuredPost?: string = "";
    image?: string = "";
    state?: string = "";
    street?: string = "";
    city?: string = "";
    website?: string = "";
    hours?: string = "";


    constructor(init?: IBusinessFormValues) {
        if (init && init.date){
            init.time = init.date
        }
        Object.assign(this, init);
    } 
}

export interface ILikedBusiness {
    username: string;
    displayName: string;
    image: string;
    isHost: boolean;
    following?: boolean;
}